### PR TITLE
[BEAM-213] Fix README to use refactored package names and use AutoService for Registrar

### PR DESCRIPTION
**Pull request by amitsela**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by amitsela** [comment]
Replacing #226  because of the git mess I made there.
Using AutoService is more "Beamish" and got rid of a bunch of classes, so it looks good - Thanks @lukecwik and @dhalperi 
Since this PR is completely different (except for the README part), please review, Thanks!

File: runners/spark/src/test/java/org/apache/beam/runners/spark/SparkRunnerRegistrarTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import java.util.ServiceLoader;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Test {@link SparkRunnerRegistrar}.
+ */
+public class SparkRunnerRegistrarTest {
```
Comment (lukecwik) [inline runners/spark/src/test/java/org/apache/beam/runners/spark/SparkRunnerRegistrarTest.java]: Please annotate class with:
@RunWith(JUnit4.class)

File: runners/spark/src/test/java/org/apache/beam/runners/spark/SparkRunnerRegistrarTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  @Test
+  public void testOptions() {
+    assertEquals(
+	    ImmutableList.of(SparkPipelineOptions.class, SparkStreamingPipelineOptions.class),
```
Comment (lukecwik) [inline runners/spark/src/test/java/org/apache/beam/runners/spark/SparkRunnerRegistrarTest.java]: fix whitespace alignment

File: runners/spark/src/test/java/org/apache/beam/runners/spark/SparkRunnerRegistrarTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    assertEquals(ImmutableList.of(SparkPipelineRunner.class),
+        new SparkRunnerRegistrar.Runner().getPipelineRunners());
+  }
+
+  @Test
+  public void testServiceLoaderForOptions() {
+    for (PipelineOptionsRegistrar registrar :
+        Lists.newArrayList(ServiceLoader.load(PipelineOptionsRegistrar.class).iterator())) {
+      if (registrar instanceof SparkRunnerRegistrar.Options) {
+      	return;
```
Comment (lukecwik) [inline runners/spark/src/test/java/org/apache/beam/runners/spark/SparkRunnerRegistrarTest.java]: add one space

File: runners/spark/src/test/java/org/apache/beam/runners/spark/SparkRunnerRegistrarTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    }
+    fail("Expected to find " + SparkRunnerRegistrar.Options.class);
+  }
+
+  @Test
+  public void testServiceLoaderForRunner() {
+    for (PipelineRunnerRegistrar registrar :
+        Lists.newArrayList(ServiceLoader.load(PipelineRunnerRegistrar.class).iterator())) {
+      if (registrar instanceof SparkRunnerRegistrar.Runner) {
+      	return;
```
Comment (lukecwik) [inline runners/spark/src/test/java/org/apache/beam/runners/spark/SparkRunnerRegistrarTest.java]: add one space

**Comment by lukecwik** [comment]
Minor fix ups and then LGTM

**Comment by dhalperi** [comment]
R: @lukecwik 

(Nothing to add here.)

**Pull request closed without merging**